docs/docs/topics/infrahub-yml.mdx

```diff
@@ -30,7 +30,18 @@ To help with the development process of a repository configuration file, you can
 
 ## GraphQL query {#graphql-query}
 
-This is the easiest type of object to link to Infrahub through an external repository because they do not need to be specified in the `.infrahub.yml` file. When Infrahub creates a Repository or pulls changes from the associated external repository, it will scan all the files in the repository and save any that have a `.gql` extension as GraphQLQuery objects in Infrahub.
+GraphQL queries could be defined in file(s) with a `.gql` extension. Then queries must also be explicitly identified in the `.infrahub.yml` file under `queries`.
+
+<details>
+  <summary>Example</summary>
+
+  ```yaml
+  queries:
+    - name: topology_info
```

Comment:
do you need to indicated the file name or the query name here ?

if I have a file toto.gql with a query tata { xxx } which name should I use ?

Reply:
File name goes to the line below. I believe the name here is the name as per in the UI (I don't think it needs to match with the query name inside the file, I haven't try to put multiple queries in the same file either ...)

Reply:
Yes the name is the name of the query within Infrahub. Previously this would just match the filename with the .gql extension removed.